server/src/main/java/org/opensearch/index/store/FsDirectoryFactory.java

```diff
@@ -104,15 +100,8 @@ protected Directory newFSDirectory(Path location, LockFactory lockFactory, Index
                 }
             case MMAPFS:
                 return setPreload(new MMapDirectory(location, lockFactory), lockFactory, preLoadExtensions);
+            // simplefs was removed in Lucene 9; support for enum is maintained for bwc
             case SIMPLEFS:
```

Comment:
Should we throw the exception here that `SIMPLEFS` is not supported anymore?

Reply:
Maybe add an `assert` instead as an invariant?

Reply:
It's not needed here since this is for bwc only.

Reply:
The deprecation logging is already handled in [MetadataUpdateSettingsService](https://github.com/opensearch-project/OpenSearch/blob/77d9b643b291b56d503abd84778ad533d0e98076/server/src/main/java/org/opensearch/cluster/metadata/MetadataUpdateSettingsService.java#L288)